src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -252,11 +247,25 @@ public void CopyRuntimeSettingsTo(JsonPropertyInfo other)
             other._escapedName = _escapedName;
         }
 
+        // Create a property that is either ignored at run-time. It uses typeof(int) in order to prevent any
+        // issues with unsupported types and helps ensure we don't accidently (de)serialize it.
+        public static JsonPropertyInfo CreateIgnoredPropertyPlaceholder(PropertyInfo propertyInfo, JsonSerializerOptions options)
+        {
+            JsonPropertyInfo jsonPropertyInfo = new JsonPropertyInfoNotNullable<int, int, int>();
+            jsonPropertyInfo.PropertyInfo = propertyInfo;
+            jsonPropertyInfo.DeterminePropertyName(options);
+
+            Debug.Assert(!jsonPropertyInfo.ShouldDeserialize);
+            Debug.Assert(!jsonPropertyInfo.ShouldSerialize);
+
+            return jsonPropertyInfo;
+        }
+
         public abstract object GetValueAsObject(object obj);
 
-        public TAttribute GetAttribute<TAttribute>() where TAttribute : Attribute
+        public static TAttribute GetAttribute<TAttribute>(PropertyInfo propertyInfo) where TAttribute : Attribute
         {
-            return (TAttribute)PropertyInfo?.GetCustomAttribute(typeof(TAttribute), inherit: false);
+            return (TAttribute)propertyInfo?.GetCustomAttribute(typeof(TAttribute), inherit: false);
```

Comment:
What motivated this change?

Reply:
The method needed to be called from outside the PropertyInfo class.